dts/x86/arduino_101.dts

```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_curie.dtsi"
+
+/ {
+	model = "Arduino 101";
+	compatible = "intel,quark";
+
+	aliases {
+		uart_0 = &uart0;
+		uart_1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart1;
+	};
+};
+
+&uart0 {
+	status = "ok";
+	current-speed = <1000000>;
```

Comment:
I believe HW flow control would come from Kconfig, if we want to do this via DT, we should look to see if the 'uart-has-rtscts' property as defined by Linux Documentation/devicetree/bindings/serial/serial.txt makes sense